### PR TITLE
Updates to CXX flags and removal of cmake checks less than 3.16.

### PR DESCRIPTION
**Pull request by larshg**
Move check for cuda up to be able to use CUDA_VERSION_STRING.
Disable Global optimization if CUDA is version 10 or 11 as they seems to be failing with their thrust version

Update how CMAKE_CXX_FLAGS_DEFAULT is set as windows default values has changed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -139,19 +147,25 @@ endif()
...
+
   if("${CMAKE_CXX_FLAGS}" STREQUAL "${CMAKE_CXX_FLAGS_DEFAULT}")
+    
     string(APPEND CMAKE_CXX_FLAGS " /fp:precise ${SSE_FLAGS} ${AVX_FLAGS}")
+    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xcompiler=/bigobj")
+
+    set(PCL_USE_GLOBAL_OPTIMIZATION TRUE)
+    if(CUDA_FOUND)
+      if(${CUDA_VERSION_STRING} VERSION_GREATER_EQUAL "10.0" AND ${CUDA_VERSION_STRING} VERSION_LESS "12.0")
+        set(USE_GLOBAL_OPTIMIZATION FALSE)
```
Comment (mvieth) [inline CMakeLists.txt:159]: Is this meant to be `PCL_USE_GLOBAL_OPTIMIZATION`? Should the user be allowed to change `PCL_USE_GLOBAL_OPTIMIZATION`, or does that not make sense?

Comment (larshg) [reply CMakeLists.txt]: Yes, indeed it was. Thanks

**Comment by larshg** [comment]
I think it already possible to change CMAKE_MSVC_CODE_LINK_OPTIMIZATION - though I don't know how. But it wasn't my intention to make PCL_USE_GLOBAL_OPTIMIZATION changeable from user side.

**Comment by larshg** [comment]
> I think it already possible to change CMAKE_MSVC_CODE_LINK_OPTIMIZATION - though I don't know how. But it wasn't my intention to make PCL_USE_GLOBAL_OPTIMIZATION changeable from user side.

CMAKE_MSVC_CODE_LINK_OPTIMIZATION  is already an cmake option that we provide 😄

**Pull request merged**